### ISSUE TITLE
keeps showing Failed to decode block body, error=EOF

### ISSUE THREAD

**Issue by bitrocks** [ethereum/go-ethereum#24158]
#### System information

Geth version: 1.10.14-stable
OS & Version:  Linux

#### Expected behaviour

Syncing blocks to the latest

#### Actual behaviour

Showing `Failed to decode block body`: 

```
INFO [12-26|11:14:07.867] Initialised chain configuration          config="{ChainID: 1 Homestead: 1150000 DAO: 1920000 DAOSupport: true EIP150: 2463000 EIP155: 2675000 EIP158: 2675000 Byzantium: 4370000 Constantinople: 7280000 Petersburg: 7280000 Istanbul: 9069000, Muir Glacier: 9200000, Berlin: 12244000, London: 12965000, Arrow Glacier: 13773000, MergeFork: <nil>, Engine: ethash}"
INFO [12-26|11:14:08.167] Disk storage enabled for ethash caches   dir=/data/chain-data/eth/geth/ethash count=3
INFO [12-26|11:14:08.167] Disk storage enabled for ethash DAGs     dir=/home/cryptodog/.ethash          count=2
INFO [12-26|11:14:08.167] Initialising Ethereum protocol           network=1 dbversion=8
INFO [12-26|11:14:08.365] Loaded most recent local header          number=13,798,950 hash=c51716..9a2441 td=36,501,196,317,498,564,696,790 age=1w5d15h
INFO [12-26|11:14:08.365] Loaded most recent local full block      number=13,798,950 hash=c51716..9a2441 td=36,501,196,317,498,564,696,790 age=1w5d15h
INFO [12-26|11:14:08.365] Loaded most recent local fast block      number=13,798,950 hash=c51716..9a2441 td=36,501,196,317,498,564,696,790 age=1w5d15h
INFO [12-26|11:14:08.365] Loaded last fast-sync pivot marker       number=12,785,936
WARN [12-26|11:14:08.365] Enabling snapshot recovery               chainhead=13,798,950 diskbase=13,800,642
WARN [12-26|11:14:08.369] Loaded snapshot journal                  diskroot=171df4..4a77a1 diffs=unmatched
WARN [12-26|11:14:08.369] Snapshot is not continuous with chain    snaproot=171df4..4a77a1 chainroot=ebd7fd..c9ef08
INFO [12-26|11:14:08.370] Loaded local transaction journal         transactions=0 dropped=0
INFO [12-26|11:14:08.370] Regenerated local transaction journal    transactions=0 accounts=0
INFO [12-26|11:14:08.370] Gasprice oracle is ignoring threshold set threshold=2
WARN [12-26|11:14:08.370] Sanitizing invalid gasprice oracle max header history provided=0 updated=1
WARN [12-26|11:14:08.370] Sanitizing invalid gasprice oracle max block history provided=0 updated=1
WARN [12-26|11:14:08.370] Old unclean shutdowns found              count=3
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-07-08T13:10:18+0000 age=5mo2w6d
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-07-08T15:00:33+0000 age=5mo2w6d
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-07-12T15:27:06+0000 age=5mo2w2d
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-07-15T04:10:58+0000 age=5mo2w7h
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-07-25T10:03:16+0000 age=5mo4d1h
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-07-28T15:24:02+0000 age=5mo19h50m
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-08-27T15:19:11+0000 age=4mo19h54m
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-12-25T05:47:53+0000 age=1d5h26m
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-12-26T01:55:36+0000 age=9h18m32s
WARN [12-26|11:14:08.370] Unclean shutdown detected                booted=2021-12-26T10:47:25+0000 age=26m43s
...
INFO [12-26|11:15:55.143] Imported new chain segment               blocks=90  txs=15401 mgas=1401.691 elapsed=8.007s mgasps=175.043 number=13,799,702 hash=c6c21b..d16319 age=1w5d12h   dirty=534.44MiB
INFO [12-26|11:16:03.183] Imported new chain segment               blocks=96  txs=17645 mgas=1494.357 elapsed=8.039s mgasps=185.871 number=13,799,798 hash=1f3ea1..6785fb age=1w5d12h   dirty=561.35MiB
INFO [12-26|11:16:03.301] Imported new chain segment               blocks=2   txs=295   mgas=22.299   elapsed=117.993ms mgasps=188.984 number=13,799,800 hash=39c26e..fe89dd age=1w5d12h   dirty=562.99MiB
WARN [12-26|11:16:03.599] Failed to decode block body              block=11,305,880 error=EOF
INFO [12-26|11:16:03.727] Importing heavy sidechain segment        blocks=902 start=13,799,801 end=13,800,702
WARN [12-26|11:16:03.753] Failed to decode block body              block=11,305,881 error=EOF
WARN [12-26|11:16:04.120] Failed to decode block body              block=11,306,102 error=EOF
WARN [12-26|11:16:04.505] Failed to decode block body              block=11,306,341 error=EOF
INFO [12-26|11:16:11.782] Imported new chain segment               blocks=37  txs=6722  mgas=581.545  elapsed=8.054s    mgasps=72.201  number=13,799,837 hash=a81b3f..def40f age=1w5d11h   dirty=574.22MiB
WARN [12-26|11:16:16.980] Failed to decode block body              block=11,429,333 error=EOF
WARN [12-26|11:16:17.295] Failed to decode block body              block=11,429,334 error=EOF
WARN [12-26|11:16:17.452] Failed to decode block body              block=11,429,335 error=EOF
WARN [12-26|11:16:17.766] Failed to decode block body              block=11,429,355 error=EOF
WARN [12-26|11:16:17.929] Failed to decode block body              block=11,429,356 error=EOF
WARN [12-26|11:16:18.242] Failed to decode block body              block=11,429,437 error=EOF
WARN [12-26|11:16:18.399] Failed to decode block body              block=11,429,438 error=EOF
WARN [12-26|11:16:18.715] Failed to decode block body              block=11,429,524 error=EOF
WARN [12-26|11:16:19.029] Failed to decode block body              block=11,429,525 error=EOF
WARN [12-26|11:16:19.186] Failed to decode block body              block=11,429,526 error=EOF
WARN [12-26|11:16:19.501] Failed to decode block body              block=11,429,589 error=EOF
WARN [12-26|11:16:19.667] Failed to decode block body              block=11,429,590 error=EOF
INFO [12-26|11:16:19.851] Imported new chain segment               blocks=56  txs=9604  mgas=857.564  elapsed=8.069s    mgasps=106.278 number=13,799,893 hash=27f55f..ece190 age=1w5d11h   dirty=594.01MiB
WARN [12-26|11:16:19.981] Failed to decode block body              block=11,429,612 error=EOF
WARN [12-26|11:16:20.138] Failed to decode block body              block=11,429,613 error=EOF
```

Does it mean the chain state is corrupted?

#### Backtrace

When submitting logs: please submit them as text and not screenshots.

**Comment by ligi** [comment]
Did you have any unclean shutdown? Can you provide more logs (from the beginning?)

**Comment by MariusVanDerWijden** [comment]
So this error only happens if the bodies are non-existent in the database.
This should *never* happen. Since it does happen, it looks like one of the unclean shutdowns destroyed your database. 
Only way forward is to delete the db and resync (you can still retain the ancient folder)

**Comment by pokrovskyy** [comment]
I stumbled upon this today. External volumes where my ancient DB and main DB were stored, got abruptly disconnected leaving both DBs in unclean state. Typically, I only delete the main DB and let it resync (like @MariusVanDerWijden  suggested), however this time it did not help and I ended up with same error as described. I realized that possibly the ancient DB also got corrupted - only after full resync (with both DBs deleted) it got resolved
